Rust/src/rcf.rs

```diff
@@ -63,7 +72,20 @@ pub trait RCF {
         normalizer: fn(f64, usize) -> f64,
     ) -> f64;
 
-    fn conditional_field(&self, positions: &[usize], point: &[f32], centrality: f64) -> Vec<usize> {
+
+    fn impute_missing_values(&self, positions: &[usize], point: &[f32]) -> Vec<f32> {
+        let mut list = self.conditional_field(positions,point,1.0);
+        let mut answer = vec![0.0f32;positions.len()];
+        for i in 0..positions.len() {
+            list.sort_by(|a, b| a[i].partial_cmp(&b[i]).unwrap());
```

Comment:
Does this need to be sorted again during each iteration?

Reply:
Well, it is the median of multiple dimensions and so the ordering can be different. One advantage is of having the sort() in the loop is that if there are correlations then there is a chance that the second sort() performs less work. One can also project to 1D and sort those ...

Reply:
I see, I missed the fact that we're sorting by different values in the vector on each iteration.